homeassistant/components/weather/__init__.py

```diff
@@ -222,78 +394,105 @@ def state_attributes(self):
             data[ATTR_WEATHER_OZONE] = ozone
 
         if (pressure := self.pressure) is not None:
-            if (unit := self.pressure_unit) is not None:
-                pressure = round(
-                    self.hass.config.units.pressure(pressure, unit), ROUNDING_PRECISION
+            with suppress(ValueError):
+                float(pressure)
+                value_pressure = UNIT_CONVERSIONS[CONF_PRESSURE_UOM](
+                    pressure, self.native_pressure_unit, self.pressure_unit
```

Comment:
What happens if either `self.native_pressure_unit` or `self.pressure_unit` is `None`?
I think we need to handle that during a migration period.